geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -172,29 +275,112 @@ def before_create(table, bind, **kw):
                     col.type, "spatial_index", False
                 ):
                     table.info["_after_create_indexes"].append(idx)
-    _setup_dummy_type(table, gis_cols)
+
+    if db_is_gpkg:
+        if len(gis_cols) > 1:
+            raise ValueError(
+                "Only one geometry column is allowed for a table stored in a GeoPackage."
+            )
+        elif len(gis_cols) == 1:
+            col = gis_cols[0]
+            srid = col.type.srid
+
+            if col.type.geometry_type is None:
+                col.type.geometry_type = "GEOMETRY"
+
+            # Add the SRID of the table in 'gpkg_spatial_ref_sys' if this table exists
+            if (
+                bind.execute(text("""PRAGMA main.table_info("gpkg_spatial_ref_sys");""")).fetchall()
+                and not bind.execute(
+                    text(
+                        """SELECT COUNT(*) FROM gpkg_spatial_ref_sys WHERE srs_id = {};""".format(
+                            srid
+                        )
+                    )
+                ).scalar()
+            ):
+                bind.execute(text("SELECT gpkgInsertEpsgSRID({})".format(srid)))
+            # table.columns = table.info.pop("_saved_columns")
+        _setup_dummy_type(table, gis_cols, gpkg=True)
+    else:
+        _setup_dummy_type(table, gis_cols, gpkg=False)
 
 
 def after_create(table, bind, **kw):
     """Handle spatial indexes during the after_create event."""
     dialect = bind.dialect
     dialect_name = dialect.name
-
-    table.columns = table.info.pop("_saved_columns")
+    db_is_gpkg = is_gpkg(bind)
+
+    if db_is_gpkg:
+        for col in table.columns:
+            # Add the managed Geometry columns with gpkgAddGeometryColumn()
+            if _check_spatial_type(col.type, Geometry, dialect) and check_management(
+                col, dialect_name
+            ):
+                col.type = col._actual_type
+                del col._actual_type
+                dimension = get_col_dim(col)
+                has_z = "Z" in dimension
+                has_m = "M" in dimension
+
+                bind.execute(
+                    text(
+                        """INSERT INTO gpkg_contents
+                        VALUES (
+                            '{}',
+                            'features',
+                            NULL,
```

Comment:
- QGis and fiona require an identifier; as every table can only have 1 geometry column I suggest putting the table name over here.
- The desciption should be an empty string, not NULL

Reply:
Ok, let's make it compatible with QGIS, it makes sense.

Reply:
I made these changes and now it seems very close to a GeoPackage created by QGIS. Does it look good to you?